// ngx-show-hide-password/esm2020/lib/show-hide-password.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { FontAwesomeModule } from '@fortawesome/angular-fontawesome';
import { ShowHideTriggerDirective } from './show-hide-trigger.directive';
import { ShowHideInputDirective } from './show-hide-input.directive';
import { ShowHideStatusDirective } from './show-hide-status.directive';
import { ShowHidePasswordComponent } from './show-hide-password.component';
import * as i0 from "@angular/core";
export class ShowHidePasswordModule {
}
ShowHidePasswordModule.ɵfac = function ShowHidePasswordModule_Factory(t) { return new (t || ShowHidePasswordModule)(); };
ShowHidePasswordModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: ShowHidePasswordModule });
ShowHidePasswordModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [CommonModule, FormsModule, FontAwesomeModule] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(ShowHidePasswordModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, FormsModule, FontAwesomeModule],
                exports: [
                    ShowHidePasswordComponent,
                    ShowHideTriggerDirective,
                    ShowHideInputDirective,
                    ShowHideStatusDirective
                ],
                declarations: [
                    ShowHidePasswordComponent,
                    ShowHideTriggerDirective,
                    ShowHideInputDirective,
                    ShowHideStatusDirective
                ]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(ShowHidePasswordModule, { declarations: [ShowHidePasswordComponent,
        ShowHideTriggerDirective,
        ShowHideInputDirective,
        ShowHideStatusDirective], imports: [CommonModule, FormsModule, FontAwesomeModule], exports: [ShowHidePasswordComponent,
        ShowHideTriggerDirective,
        ShowHideInputDirective,
        ShowHideStatusDirective] }); })();
