// ngx-show-hide-password/esm2020/lib/show-hide-trigger.directive.mjs
import { Directive, HostListener, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "./show-hide.service";
export class ShowHideTriggerDirective {
    constructor(service, errorHandler) {
        this.service = service;
        this.errorHandler = errorHandler;
    }
    onClick() {
        if (this.showHideTrigger) {
            this.service.toggleShow(this.showHideTrigger);
        }
        else {
            this.errorHandler.handleError(new Error(`Status can not be changed without [id].`));
        }
    }
}
ShowHideTriggerDirective.ɵfac = function ShowHideTriggerDirective_Factory(t) { return new (t || ShowHideTriggerDirective)(i0.ɵɵdirectiveInject(i1.ShowHideService), i0.ɵɵdirectiveInject(i0.ErrorHandler)); };
ShowHideTriggerDirective.ɵdir = /*@__PURE__*/ i0.ɵɵdefineDirective({ type: ShowHideTriggerDirective, selectors: [["", "showHideTrigger", ""]], hostBindings: function ShowHideTriggerDirective_HostBindings(rf, ctx) { if (rf & 1) {
        i0.ɵɵlistener("click", function ShowHideTriggerDirective_click_HostBindingHandler() { return ctx.onClick(); });
    } }, inputs: { showHideTrigger: "showHideTrigger" } });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(ShowHideTriggerDirective, [{
        type: Directive,
        args: [{
                // eslint-disable-next-line @angular-eslint/directive-selector
                selector: '[showHideTrigger]'
            }]
    }], function () { return [{ type: i1.ShowHideService }, { type: i0.ErrorHandler }]; }, { showHideTrigger: [{
            type: Input
        }], onClick: [{
            type: HostListener,
            args: ['click']
        }] }); })();
