// ngx-show-hide-password/esm2020/lib/show-hide-status.directive.mjs
/* eslint-disable @angular-eslint/directive-selector */
import { Directive, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "./show-hide.service";
const defaultConfig = {
    show: 'visibility',
    hide: 'visibility_off',
    materialIcon: false,
    id: undefined,
};
export class ShowHideStatusDirective {
    constructor(service, el, renderer, errorHandler) {
        this.service = service;
        this.el = el;
        this.renderer = renderer;
        this.errorHandler = errorHandler;
        this.config = defaultConfig;
    }
    set showHideStatus(config) {
        this.init(config);
    }
    init(config) {
        this.config = {
            ...defaultConfig,
            ...config,
        };
        if (this.config.id) {
            this.subscription = this.service
                .getObservable(this.config.id)
                .subscribe((show) => this.updateStatus(show));
        }
        else {
            this.errorHandler.handleError(new Error(`Status can not be set without [id].`));
        }
    }
    updateStatus(show) {
        if (this.config.materialIcon) {
            this.renderer.setProperty(this.el.nativeElement, 'innerHTML', show ? this.config.hide : this.config.show);
        }
        else {
            this.renderer.removeClass(this.el.nativeElement, (!show ? this.config.hide : this.config.show) ?? '');
            this.renderer.addClass(this.el.nativeElement, (show ? this.config.hide : this.config.show) ?? '');
        }
    }
    ngOnDestroy() {
        if (this.subscription) {
            this.subscription.unsubscribe();
        }
    }
}
ShowHideStatusDirective.ɵfac = function ShowHideStatusDirective_Factory(t) { return new (t || ShowHideStatusDirective)(i0.ɵɵdirectiveInject(i1.ShowHideService), i0.ɵɵdirectiveInject(i0.ElementRef), i0.ɵɵdirectiveInject(i0.Renderer2), i0.ɵɵdirectiveInject(i0.ErrorHandler)); };
ShowHideStatusDirective.ɵdir = /*@__PURE__*/ i0.ɵɵdefineDirective({ type: ShowHideStatusDirective, selectors: [["", "showHideStatus", ""]], inputs: { showHideStatus: "showHideStatus" } });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(ShowHideStatusDirective, [{
        type: Directive,
        args: [{
                selector: '[showHideStatus]',
            }]
    }], function () { return [{ type: i1.ShowHideService }, { type: i0.ElementRef }, { type: i0.Renderer2 }, { type: i0.ErrorHandler }]; }, { showHideStatus: [{
            type: Input
        }] }); })();
